test/e2e/scheduler_predicates.go

```diff
@@ -900,16 +886,37 @@ func getRequestedCPU(pod v1.Pod) int64 {
 	return result
 }
 
-func waitForScheduler() {
-	// Wait a bit to allow scheduler to do its thing
-	// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
-	framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
-	time.Sleep(10 * time.Second)
+func waitForScheduler(f *framework.Framework, PodName string, expectSuccess bool) {
+	framework.Logf("Waiting for scheduler...")
+	action := func() error { // No action before waiting for Scheduler.
+		return nil
```

Comment:
Thanks, Connor! I also realized the race issue when creating the PR, but the reason that I created this method is that at line 732, the test waits for scheduler after it removes a taint from a node. No pod is created in this case, but the race condition still exists.
I guess I can create another method similar to `createPausePodAndWaitForScheduler` that removes a taint and waits for scheduler. A better way is to write a function like `takeActionAndWaitForScheduler` that receives another function as one of its arguments, uses the function as `action` and then waits for scheduler. In our example, the action will be the removal of a taint from a node. What do you think?